ambry-store/src/main/java/com.github.ambry.store/PersistentIndex.java

```diff
@@ -1063,6 +1063,47 @@ long getLogUsedCapacity() {
     return getAbsolutePositionInLogForOffset(getCurrentEndOffset(indexSegments), indexSegments);
   }
 
+  /**
+   * @return absolute end position (in bytes) of latest PUT record when this method is invoked.
+   * @throws StoreException
+   */
+  long getAbsoluteEndPositionOfLastPut() throws StoreException {
+    IndexValue indexValueOfLastPut = null;
+    // 1. go through keys in journal in reverse order to see if there is a PUT index entry associated with the key
+    List<JournalEntry> journalEntries = journal.getAllEntries();
+    ConcurrentSkipListMap<Offset, IndexSegment> indexSegments = validIndexSegments;
```

Comment:
unnecessary line?

Reply:
Reference `validIndexSegments` might be changed to point to another ConcurrentSkipListMap (which is inFluxIndexSegments) during iteration. We still use old view to finish finding the last PUT.  (Although, in my move replica design, I will disable compaction first before calling this method, I still would like to keep a "dedicated" reference pointing to the indexSegments view at the time when the method is called)